modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordBinaryManager.kt

```diff
@@ -0,0 +1,195 @@
+package com.metalbear.mirrord
+
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.progress.Task
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.SystemInfo
+import com.intellij.util.system.CpuArch
+import java.nio.charset.Charset
+import java.nio.file.Path
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.TimeUnit
+import java.net.URI
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+import java.net.URLEncoder
+import java.nio.file.Files
+import java.time.Duration
+
+/**
+ * For dynamically fetching and storing mirrord binary.
+ */
+object MirrordBinaryManager {
+    private const val cliBinary = "mirrord"
+    private const val versionEndpoint = "https://version.mirrord.dev/v1/version"
+    private const val downloadEndpoint = "https://github.com/metalbear-co/mirrord/releases/download"
+
+    /**
+     * @return version of mirrord binary
+     *
+     * @throws RuntimeException if `--version` command failed
+     */
+    private fun getVersion(cliPath: String): String {
+        val child = Runtime.getRuntime().exec(arrayOf(cliPath, "--version"))
+
+        val result = child.waitFor()
+        if (result != 0) {
+            MirrordLogger.logger.debug("`mirrord --version` failed with code $result")
+            throw RuntimeException("failed to get mirrord version")
```

Comment:
imo make this fallible so if we have weird mirrord in path we ignore and not error